run.py

```diff
@@ -537,6 +538,21 @@ def multi_synthesis(
             background=BackgroundTask(delete_file, f.name),
         )
 
+    @app.post(
+        "/optimal_pitch",
+        response_model=StylePitchRange,
+        tags=["その他"],
+        summary="指定したスタイルに対して最適なピッチ範囲を得る",
+    )
+    def optimal_pitch(style_id: StyleId) -> StylePitchRange:
```

Comment:
### APIとデータ構造について

ピッチ範囲を取得できる`/optimal_pitch`APIを実装していますが、これはAPIを足すのではなく、スタイル関連情報がまとまっているところに足す形だとより扱いやすいかもしれません。
例えば`SpeakerStyle`内などです。
ただそれをするにはコードを読んでみた感じかなり改革が必要そうでした。でも良いAPI設計のためにやるべきだと感じています。
改革はプルリクエスト範囲から外れているので興味なければ苦痛かもしれません。もし興味あれば以下の議論に参加して頂ければ。

---

@y-chan さんと @tarepan さんに相談です。
たぶんピッチ範囲を格納するのは、StyleType等が保存されてる`SpeakerStyle`内が最適だと思うがどうでしょうか。
ただ、もし仮に`SpeakerStyle`にピッチ範囲を含める場合は問題があります。
`SpeakerStyle`は`CoreSpeaker`内にあるのですが、`CoreSpeaker`はコアの読み込みから利用されているため、変更できなくなっています。
そもそもAPIと無関係であるべきな`CoreSpeaker`モデルがここに定義されてることが変で、コア用のデータ構造はコア読み込みの部分だけで閉じているべきに感じました。
なので`metas/Metas.py`内から`CoreSpeaker`の定義をやめ、`Speaker`直下に`.name`や`.speaker_uuid`や`.supported_features`を定義するのとかどうでしょうか。
たぶんAPIの返り値は一切変えずに`CoreSpeaker`定義を失くせると思っています。

この健全化作業を @tarepan さんにお任せしても良いでしょうか🙇
その変更のあと、 @Patchethium さんがこのプルリクエスト内で今の`SpeakerStyle`内に`.pitch_range`を追加するというのはどうでしょうか。

<details><summary>english</summary>

### About the API and data structure

We are implementing an `/optimal_pitch` API that can retrieve the pitch range, but instead of adding this as a separate API, it might be more manageable to add it to a place where style-related information is aggregated, such as inside `SpeakerStyle`.
However, from what I've seen reading the code, this might require quite a bit of reform. But I feel it's worth doing for good API design.
This reform might be outside the scope of a pull request, so it could be a pain if you're not interested. But if you are interested, I would appreciate your participation in the following discussion.

---

@y-chan and @tarepan
Perhaps the best place to store the pitch range would be inside `SpeakerStyle`, where `StyleType`, etc., are stored, but there are issues with that.
`SpeakerStyle` is inside `CoreSpeaker`, but `CoreSpeaker` is being used from the core load, making it unchangeable.
It's odd that the `CoreSpeaker` model, which should be unrelated to the API, is defined here. I feel that the data structure for the core should be confined to just the core loading part.
So, how about stopping the definition of `CoreSpeaker` inside `metas/Metas.py` and defining `.name`, `.speaker_uuid`, `.supported_features`, etc., directly under `Speaker`?
I believe we can remove the definition of `CoreSpeaker` without changing any API responses at all.

Could we leave this rationalization task to @tarepan 🙇
After that change, how about @Patchethium adding `.pitch_range` to the current `SpeakerStyle` within this pull request?

</details>

Reply:
> ピッチ範囲を格納するのは ... `SpeakerStyle` 内が最適  

👍️  
同意です。  
意味論からすると `SpeakerStyle` 内に配置するのが妥当と考えます。  

> APIと無関係であるべきな`CoreSpeaker`モデルがここに定義されてることが変で、コア用のデータ構造はコア読み込みの部分だけで閉じているべき  

👍️  
同意です。  
現状は、`list[Speaker]` が VV API に露出する型であり、`CoreSpeaker` が Core API を包む型となっています。  
`Speaker` の定義時に `CoreSpeaker` 継承を用いた結果 VV API と Core API が相互依存化し、CoreAPI が不変だからピッチ情報追加 VV API 変更ができない、という状況に陥っています。   

> `CoreSpeaker`の定義をやめ、`Speaker`直下に`.name`や ... を定義  

👍️  
同意です。  
`class Speaker(CoreSpeaker, EngineSpeaker):` による VV API - Core API 相互依存がこれで解けます。  

> 健全化作業を @ tarepan さんにお任せしても良いでしょうか

👍️  
承りました。複数 steps / PRs で構成されるリファクタリングをおこないます。  
影響範囲が複数ファイルにまたがるため、いま健全化すると既存 PRs とコンフリクト祭りします。他 PRs が review & merge され着手可能になった段階で、@hiroshiba さんから @ tarepan へ改めてメンションをお願いします。着手可能メンションを受け次第、実装に着手します。  

--------------------------------------

ピッチ範囲情報の出元に関するコメントです。  
ENGINE の `metas.json` ではなく CORE から取得すべきだと考えます。  
なぜならピッチ範囲情報と並列関係にある `name` / `id` / `type` が CORE から取得する情報であるからです。

Reply:
私はコアに関する知識が一切ないのでなんとも言えないですが、スタイル情報を持っているコア側から取るのは一番自然なのは同意です。もし @tarepan さん）が実装してくればこれで一番良いと思います。
それでは一応[ここ](https://github.com/VOICEVOX/voicevox_resource/pull/60)をcloseにして、コア側の健全化作業を待つことにします。

Reply:
遅くなりました！！ 🙇 

@tarepan ありがとうございます！！

> 他 PRs が review & merge され着手可能になった段階で、@Hiroshiba さんから @ tarepan へ改めてメンションをお願いします。

承知しました！と言いたいところなのですが、ちょっとどのタイミングならいけやすいかPR全部把握しているわけではないのでわからないというのが正直なところです。。 🙇 
今ぱっとPRのタイトルを見て回った感じ、API Routerの変更とテストの追加が主だと感じました。
このタスクはWebAPI周り・コア周りのSpeakerデータ構造のタスクなので、意外と今衝突していないかもしれません。

Reply:
@Patchethium  
お待たせしました。#1202 がマージされ、エンジンのデータ構造が整理されました。  
ピッチ範囲をエンジンの `metas.json` に配置するのであれば実装可能かと思います（コアの `metas.json` に置くかどうかはメンテナ判断かと思います）。